tests_tf/test_attacks.py

```diff
@@ -44,6 +57,175 @@ def test_generate_np(self):
         # test perturbation norm
         self.assertTrue(np.allclose(perturbation_norm, self.attack.eps))
 
+class TestFastGradientMethod(CleverHansTest):
+    def setUp(self):
+        super(TestFastGradientMethod, self).setUp()
+        import tensorflow as tf
+
+        # The world's simplest neural network
+        def my_model(x):
+            W1 = tf.constant([[1.5, .3], [-2, 0.3]], dtype=tf.float32)
+            h1 = tf.nn.sigmoid(tf.matmul(x, W1))
+            W2 = tf.constant([[-2.4, 1.2], [0.5, -2.3]], dtype=tf.float32)
+            res = tf.nn.softmax(tf.matmul(x, W2))
+            return res
+
+        self.sess = tf.Session()
+        self.model = my_model
+        self.attack = FastGradientMethod(self.model, sess=self.sess)
+
+    def test_generate_np_gives_adversarial_example(self):
+        x_val = np.random.rand(100, 2)
+        x_val = np.array(x_val, dtype=np.float32)
+
+        for ord in [np.inf, 1, 2]:
+            x_adv = self.attack.generate_np(x_val, eps=.5, ord=ord,
+                                            clip_min=-5, clip_max=5)
+
+            if ord == np.inf:
+                assert np.allclose(np.max(np.abs(x_adv-x_val), axis=1), 0.5)
```

Comment:
My understanding is that it is better to use the assertions provided by the unittest package than `assert` in the tests. If that it is the case, would it make sense to switch all of the following assert statements?